argus/environment.py

```diff
@@ -62,11 +62,37 @@ def __init__(self, config_file, config_opts,
         self._start_commands = start_commands
         self._stop_commands = stop_commands
 
-    @staticmethod
-    def _run_commands(commands):
+    def _run_commands(self, commands):
         for command in commands:
+            self._run_command(command)
+
+    @staticmethod
+    def _run_command(command):
+        """Runs a command and returns the output.
+
+        Params:
+            command (string or list) - the command to be executed
+
+        Return value:
+            The stdout of the command as a list of string. Each string
+            represents a line of the output.
+
+        Raises:
+            TypeError if type(command) not in [list, str]
+
+        """
+        if type(command) is str:
             args = shlex.split(command)
-            subprocess.call(args, shell=False)
+        elif type(command) is list:

```

Comment:
I don't see run_command to actually be called with a list, so this check is redundant. Also, make sure to use isinstance, not type(something), since that's the idiomatic way.